File: boj/2023/0-str-230417_0512/1120.cpp
#include <cstdio>
#include <cstring>

char A[51], B[51];
int lA, lB;

int main()
{
    scanf("%s %s", A, B);
    lA = strlen(A), lB = strlen(B);

    int min_diff = 1e9, diff;
    for (int i = 0; i <= lB - lA; i++) {
        diff = 0;
        for (int j = 0; j < lA; j++) {
            if (B[i + j] - A[j])
                diff++;
        }
        if (diff < min_diff)
            min_diff = diff;
    }
    printf("%d", min_diff);
}

File: boj/2023/tags/str-230417_0512/1120.py
A, B = input().split()

min_diff = int(1e9)
for i in range(len(B) - len(A) + 1):
    diff = 0
    for j in range(len(A)):
        if A[j] != B[i + j]:
            diff += 1
    if diff < min_diff:
        min_diff = diff
print(min_diff)